vector/src/main/java/im/vector/adapters/VectorRoomSummaryAdapter.java

```diff
@@ -810,7 +806,7 @@ public View getChildView(int groupPosition, int childPosition, boolean isLastChi
         }
 
         if (null != childRoom) {
-            directChatIcon.setVisibility(mDirectChatRoomIdsList.indexOf(childRoom.getRoomId()) < 0 ? View.GONE : View.VISIBLE);
+            directChatIcon.setVisibility(!RoomUtils.isDirectChat(mMxSession, childRoom.getRoomId()) ? View.GONE : View.VISIBLE);
```

Comment:
Can you please remove the `!` in the condition and inverse the View.GONE, View.VISIBLE?